### PR TITLE
Fix attribute ids validation

### PR DESCRIPTION
**Pull request by pieterlukasse**
# What? Why?
Fix to prevent scenario described in #2595. 

Changes proposed in this pull request:
- validation script now gives ERROR when clinical attribute ID contains any special character (except for _ underscore). Special characters are still allowed in attribute name and description (which is what appears on UI).

### CODE REVIEW

**Comment by pieterlukasse** [comment]
here is a list of studies that will fail validation after this fix:

https://github.com/bsir/cBioPortal-analysis/blob/master/attribute_formatting_check.ipynb

@sandertan @ritikakundra let's discuss how to fix this in datahub.

**Comment by n1zea144** [comment]
I think this is a good idea and we should coordinate the update of data with this fix.  I think @ritikakundra should help with this and update data in datahub that is out of sync with the public portal.

**Comment by pieterlukasse** [comment]
thanks @sandertan . Yes, I would like to keep this rule since it makes our data a bit more developer friendly IMO (many developers wouldn't assume an ID to have spaces or other special characters).

**Comment by jjgao** [comment]
@pieterlukasse should we still merge this? If so, could you rebase?

File: core/src/main/scripts/importer/validateData.py [hunk 1]
```diff
@@ -1792,6 +1792,15 @@ def checkHeader(self, cols):
                     extra={'line_number': self.line_number,
                            'column_number': col_index + 1,
                            'cause': col_name})
+            # do not allow special characters in attribute names
+            if not re.match(
+                    # regex to allow only alpha numerics and _ underscore:
+                    r'^[A-Za-z0-9_]+$', col_name):
```
Comment (jjgao) [inline core/src/main/scripts/importer/validateData.py]: should we support `[^A-Za-z0-9\._-]`?

**Comment by jjgao** [comment]
@pieterlukasse @n1zea144 I still think it's a good idea to normalize attribute IDs. We can query our database and see how much data we need to fix.

File: core/src/main/scripts/importer/validateData.py (hunk 1 above)
Comment (pieterlukasse) [reply core/src/main/scripts/importer/validateData.py]: @jjgao any specific public studies you have in mind?

**Comment by jjgao** [comment]
@pieterlukasse any plan for this?

**Comment by jjgao** [comment]
I am closing this one since it's very old and #2595 has been addressed in frontend. Please open a new one if this is still relevant.

**Pull request closed without merging**